pilot/pkg/model/context.go

```diff
@@ -86,6 +87,10 @@ type Proxy struct {
 	// NOTE: DO NOT USE THIS FIELD TO CONSTRUCT DNS NAMES
 	ConfigNamespace string
 
+	// NamespaceDependencies contains a list of namespaces that should be used to configure
+	// services for this node.
+	NamespaceDependencies map[string]bool
```

Comment:
rename to ConfigNamespaceDependencies

Reply:
Note that in my PR I removed this - just hold the actual Service ( and any other object that is specific to the proxy).

Reply:
With ConfigScope this is a little complicated.

Reply:
Thinking more about this, remove this from the proxy and move it to PushContext. You need to compute this only once with the current service dependency/global config thing. We are not allowing service dependency specialization per workload yet. So, its at namespace level. which is far more easier and more efficient than computing this for *every* proxy.